include/deal.II/lac/linear_operator.h

```diff
@@ -0,0 +1,1175 @@
+ *
+ * @code
+ * Domain & operator *=(Domain::value_type);
+ * Range & operator *=(Range::value_type);
+ * @endcode
+ */
+template <typename Range, typename Domain>
+LinearOperator<Range, Domain>
+operator*(typename Range::value_type  number,
+          const LinearOperator<Range, Domain> &op)
+{
+  static_assert(
+    std::is_convertible<typename Range::value_type, typename Domain::value_type>::value,
+    "Range and Domain must have implicitly convertible 'value_type's");
+
+  LinearOperator<Range, Domain> return_op = op;
+
+  // ensure to have valid computation objects by catching number and op by
+  // value
+
+  return_op.vmult = [number, op](Range &v, const Domain &u)
+  {
+    op.vmult(v,u);
+    v *= number;
+  };
+
+  return_op.vmult_add = [number, op](Range &v, const Domain &u)
+  {
+    v /= number;
+    op.vmult_add(v,u);
+    v *= number;
+  };
+
+  return_op.Tvmult = [number, op](Domain &v, const Range &u)
+  {
+    op.Tvmult(v,u);
+    v *= number;
+  };
+
+  return_op.Tvmult_add = [number, op](Domain &v, const Range &u)
+  {
+    v /= number;
+    op.Tvmult_add(v,u);
+    v *= number;
+  };
+
+  return return_op;
+}
+
+
+/**
+ * \relates LinearOperator
+ *
+ * Scalar multiplication of a ScalarOperator object from the right.
+ *
+ * The @tref Domain and @tref Range types must implement the following
+ * <code>operator*=</code> member functions for rescaling:
+ *
+ * @code
+ * Domain & operator *=(Domain::value_type);
+ * Range & operator *=(Range::value_type);
+ * @endcode
+ */
+template <typename Range, typename Domain>
+LinearOperator<Range, Domain>
+operator*(const LinearOperator<Range, Domain> &op,
+          typename Domain::value_type  number)
+{
+  static_assert(
+    std::is_convertible<typename Range::value_type, typename Domain::value_type>::value,
+    "Range and Domain must have implicitly convertible 'value_type's");
+
+  return number * op;
+}
+
+
+/**
+ * \relates LinearOperator
+ *
+ * Returns the transpose linear operations of @ref op.
+ */
+template <typename Range, typename Domain>
+LinearOperator<Domain, Range>
+transpose_linop(const LinearOperator<Range, Domain> &op)
+{
+  LinearOperator<Domain, Range> return_op;
+
+  return_op.reinit_range_vector = op.reinit_domain_vector;
+  return_op.reinit_domain_vector = op.reinit_range_vector;
+
+  return_op.vmult = op.Tvmult;
+  return_op.vmult_add = op.Tvmult_add;
+  return_op.Tvmult = op.vmult;
+  return_op.Tvmult_add = op.vmult_add;
+
+  return return_op;
+}
+
+
+/**
+ * \relates LinearOperator
+ *
+ * Returns a LinearOperator that is the identity of the vector space
+ * @tref Range.
+ *
+ * The function takes an <code>std::function</code> object @ref exemplar as
+ * an argument to initialize the <code>reinit_range_vector</code> and
+ * <code>reinit_domain_vector</code> objects of the LinearOperator object.
+ */
+template <typename Range>
+LinearOperator<Range, Range>
+identity_linop(const std::function<void(Range &, bool)> &exemplar)
```

Comment:
That doesn't quite match the meaning of the word exemplar. An exemplar would be a vector that,if copied, would produce the kind of vector you want. Instead,  the argument  pass here is a function that allows resizinga vector to the correct size. I  think either way  is better or worse, it's just that the name doesn't match.